pkg/kubernetes/scripts/kube-client.js

```diff
@@ -442,13 +451,29 @@
                 var present = { };
                 var removed = { };
                 var i, len, frame, link, resource, key;
+                var cVersion, lVersion;
                 for (i = 0, len = drain.length; i < len; i++) {
                     resource = drain[i].object;
                     if (resource) {
                         link = resourcePath([resource]);
                         if (drain[i].type == "DELETED") {
                             delete objects[link];
                             removed[link] = resource;
+                        } else if (drain[i].checkResourceVersion) {
```

Comment:
Why don't we do this for both watches and loads? The only time I can think we may not want to do this is when loader.handle() is called explicitly from outside of the KubeClient code.